data_pipeline_api/registry/common.py

```diff
@@ -156,7 +157,13 @@ def get_remote_filesystem_and_path(
         password = storage_options.pop("password", None) or inferred_options.get("password")
         uri = (Path(inferred_options["path"]) / Path(path)).as_posix()
         fs_class = FTPFileSystem if protocol == "ftp" else SFTPFileSystem
-        return fs_class(host=inferred_options["host"], username=username, password=password, **storage_options), uri
+        fs = fs_class(host=inferred_options["host"], username=username, password=password, **storage_options)
+        if protocol == "ftp":
+            try:
+                fs.ftp.dir()
+            except TimeoutError:
+                fs.ftp.set_pasv(False)
```

Comment:
I wonder if it is worth attempting the `dir` again after changing this?

Reply:
I felt that that would be captured immediately on any use